org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/CorePlugin.java

```diff
@@ -263,7 +262,9 @@ public static Logger logger() {
     }
 
     public static PublishedGradleVersionsWrapper publishedGradleVersions() {
-        return (PublishedGradleVersionsWrapper) getInstance().publishedGradleVersionsServiceTracker.getService();
+    	BundleContext bundleContext = FrameworkUtil.getBundle(CorePlugin.class).getBundleContext();
```

Comment:
You should really use a servicetracker here see for example here how to handle such legacy cases:
https://github.com/eclipse-m2e/m2e-core/blob/25d6833f9cf78e0c423155c9d8381f0db8eaa9d0/org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/MavenPluginActivator.java#L114

Reply:
How is the tracker closed in your example?

Reply:
All services are automatically freed when the bundle stops, you can of course clear them mnaually if you like also see for a more complete example:

https://github.com/eclipse-platform/eclipse.platform.runtime/pull/28/files

Reply:
Your decisions of course, I usually try to avoid migrating useless/outdated stuff as one has to test this twice then or no one feels to change this later on.

Reply:
Thanks, I update the PR with your suggestion.